src/main/cpp/src/datetime_parser.cu

```diff
@@ -0,0 +1,453 @@
+  //   val zonedDateTime = ZonedDateTime.of(localDateTime, zoneId)
+  //   val instant = Instant.from(zonedDateTime) // main work
+  //   instantToMicros(instant)
+  // here just return a zero
+  return cudf::timestamp_us{cudf::duration_us{0L}};
+}
+
+__device__ __host__ inline bool is_digit(const char chr) {
+  return (chr >= '0' && chr <= '9');
+}
+
+__device__ __host__ inline bool is_whitespace(const char chr) {
+  switch (chr) {
+    case ' ':
+    case '\r':
+    case '\t':
+    case '\n': return true;
+    default: return false;
+  }
+}
+
+/**
+ * first trim the time zone,
+ * then format (+|-)h:mm, (+|-)hh:m or (+|-)h:m to (+|-)hh:mm
+ * Refer to: https://github.com/apache/spark/blob/v3.5.0/sql/api/src/main/scala/
+ * org/apache/spark/sql/catalyst/util/SparkDateTimeUtils.scala#L39
+ */
+__device__ __host__ cudf::string_view format_zone_id(const cudf::string_view &time_zone_id) {
+  const char *curr_ptr = time_zone_id.data();
+  const char *end_ptr = curr_ptr + time_zone_id.size_bytes();
+
+  // trim left
+  int num_of_left_white_space = 0;
+  while (curr_ptr < end_ptr && is_whitespace(*curr_ptr)) {
+    ++curr_ptr;
+    ++num_of_left_white_space;
+  }
+  // trim right
+  while (curr_ptr < (end_ptr - 1) && is_whitespace(*(end_ptr - 1))) {
+    --end_ptr;
+  }
+
+  const int length_after_trim = end_ptr - curr_ptr;
+  int state = 0;
+  char ret[] = "+00:00";     // save the formatted result
+  bool is_valid_form = true; // is one form of: (+|-)h:mm$, (+|-)hh:m$, (+|-)h:m$, (+|-)hh:mm$
+  int curr_digit_num = 0;
+  while (curr_ptr <= end_ptr && is_valid_form) {
+    char chr = *curr_ptr;
+    if (0 == state) {                                           // expect '+' or '-'
+      if (curr_ptr == end_ptr || !('+' == chr || '-' == chr)) { // get $
+        is_valid_form = false;
+      } else { // get '+' or '-'
+        ret[0] = chr;
+        state = 1;
+      }
+    } else if (1 == state) {     // exepct hour digits then ':'
+      if (curr_ptr == end_ptr) { // get $
+        is_valid_form = false;
+      } else if (is_digit(chr) && curr_digit_num < 2) { // get digit
+        ++curr_digit_num;
+        // set hh part
+        ret[1] = ret[2];
+        ret[2] = chr;
+      } else if (':' == chr && curr_digit_num > 0) { // get ':'
+        curr_digit_num = 0;
+        state = 2;
+      } else {
+        is_valid_form = false;
+      }
+    } else if (2 == state) {                            // expect minute digits then '$'
+      if (curr_ptr == end_ptr && curr_digit_num > 0) {  // get $
+        state = 3;                                      // success state
+      } else if (is_digit(chr) && curr_digit_num < 2) { // get digit
+        ++curr_digit_num;
+        // set mm part
+        ret[4] = ret[5];
+        ret[5] = chr;
+      } else {
+        is_valid_form = false;
+      }
+    }
+    ++curr_ptr;
+  }
+
+  if (3 == state) {
+    // success
+    return cudf::string_view(ret, 6);
```

Comment:
If I understand correctly, the `cudf::string_view` refers to a local variable which will be deallocated in the outside scope.

Reply:
Yes, we should be careful about this.
Done.
Now only construct cudf::string_view from Column View, and ensure cudf string_view is safe.